brats/losses.py

```diff
@@ -0,0 +1,38 @@
+import torch
+import torch.nn as nn
+
+
+class DiceLossOneClass(nn.Module):
+    """
+    Compute Dice Loss for single class.
+    The loss will be averaged across channels.
+    """
+
+    def __init__(self, epsilon: float = 1e-5):
+        """
+        Args:
+            epsilon: smooth factor to prevent division by zero
+        """
+        super(DiceLossOneClass, self).__init__()
+        self.epsilon = epsilon
+
+    def __call__(self, prediction: torch.Tensor,
+                 target: torch.Tensor) -> torch.Tensor:
+        """
+        Args:
+            prediction: output of a network, expected to be already binarized.
+                Dimensions - (Batch, Channels, Height, Width)
+            target: labels.
+                Dimensions - (Batch, Channels, Height, Width)
+
+        Returns:
+            torch.Tensor: Computed Dice Loss
+        """
```

Comment:
I think it should comply to the NCDWH, otherwise we would need to do reshaping every time we calculate a loss. I suggest making It work with NCDWH with one channel and raising an exception if the input is larger

Reply:
Added Channel dimension